architecture.md

```diff
@@ -1,11 +1,86 @@
-# MongoDB Community Operator Architecture
+# MongoDB Community Kubernetes Operator Architecture
 
-**Work in Progress**
+The MongoDB Community Kubernetes Operator is a [Custom Resource Definition](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) and a [Controller](https://kubernetes.io/docs/concepts/architecture/controller/).
 
-## Topics
+## Table of Contents
 
-* Automation Agent
-* Cluster Configuration and ConfigMap
-* MongoDB Docker Images
-* Example case: MongoDB version upgrade
+- [Cluster Configuration](#cluster-configuration)
+- [Example: MongoDB Version Upgrade](#example-mongodb-version-upgrade)
+- [MongoDB Docker Images](#mongodb-docker-images)
 
+## Cluster Configuration
+
+You create and update MongoDB resources by defining a MongoDB resource definition. When you apply the MongoDB resource definition to your Kubernetes environment, the Operator:
+
+1. Creates a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) that contains one [pod](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/) for each [replica set](https://docs.mongodb.com/manual/replication/) member. 
+1. Creates two [containers](https://kubernetes.io/docs/concepts/containers/overview/) in each pod:
+
+   - A container for the [`mongod`](https://docs.mongodb.com/manual/reference/program/mongod/index.html) process binary. </br>
+     `mongod` is the primary daemon process for the MongoDB system. It handles data requests, manages data access, and performs background management operations.
+
+   - A container for the Automation Agent. </br>
+     The Automation Agent handles configuring, stopping, and restarting the `mongod` process. The Automation Agent periodically polls the `mongod` to determine status and can deploy changes as needed. 
+1. Writes the Automation configuration as a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) and mounts it to each pod. 
+1. Initiates the Automation Agent, which in turn creates the database configuration and launches the `mongod` process according to your MongoDB resource definition.
+
+<!--
+<img src="" alt="Architecure diagram of the MongoDB Community Kubernetes Operator">
+-->
+
+This architecture maximizes use of the Automation Agent while integrating naturally with Kubernetes to produce a number of benefits.
+
+- MongoDB containers are not tied to the Operator or the Agent, so you can:
```

Comment:
It's not clear to me what this means exactly. Assuming this means you can use custom MongoDB images with the operator. Do you specify an image to pull in a config somewhere?

Reply:
@rodrigovalin Can you clarify?

Reply:
Updated.